src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -484,6 +485,9 @@ class TestWithServers(TestWithoutServers):
     :avocado: recursive
     """
 
+    # whether engines ULT stacks have been already dumped
+    dumped_engines_stacks = False
```

Comment:
Using class attributes through self seems a little bit strange for me.
From what I know, when you use class attributes you should access it through `TestWithServers.dumped_engines_stacks`.
From what I understand, you should use an instance attribute initialized in the `__init__` function.
However, it is highly possible that I am missing something as I am completely unfamiliar with this code.

Reply:
Hmm, I was not aware that such initialisation way was causing all instances to share the same single value... And my C++ days are too far now, but is this always the same for object oriented languages ? ok, will move it to __init__() method of TestWithServers class.

Reply:
Yop, for all "real" object oriented language that I know.